pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,471 @@
+		T::RoundIndex,
+		Blake2_128Concat,
+		T::AccountId,
+		T::AccountId,
+		OptionQuery,
+	>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub min_orbiter_deposit: BalanceOf<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				min_orbiter_deposit: One::one(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			assert!(
+				self.min_orbiter_deposit > Zero::zero(),
+				"Minumal orbiter deposit should be greater tham zero"
+			);
+			MinOrbiterDeposit::<T>::put(self.min_orbiter_deposit)
+		}
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_: T::BlockNumber) -> Weight {
+			// Prune old OrbiterPerRound entries
+			let current_round = CurrentRound::<T>::get();
+			if current_round > T::MaxRoundArchive::get() {
+				let round_to_prune = current_round - T::MaxRoundArchive::get();
+				OrbiterPerRound::<T>::remove_prefix(round_to_prune, None);
+				T::DbWeight::get().reads_writes(1, 1)
+			} else {
+				T::DbWeight::get().reads(1)
+			}
+		}
+	}
+
+	/// An error that can occur while executing this pallet's extrinsics.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The collator is already added in orbiters program.
+		CollatorAlreadyAdded,
+		/// This collator is not in orbiters program.
+		CollatorNotFound,
+		/// There are already too many orbiters associated with this collator.
+		CollatorPoolTooLarge,
+		/// There are more collator pools than the number specified in the parameter.
+		CollatorsPoolCountTooLow,
+		/// The minimum deposit required to register as an orbiter has not yet been included in the
+		/// onchain storage
+		MinOrbiterDepositNotSet,
+		/// This orbiter is already associated with this collator.
+		OrbiterAlreadyInPool,
+		/// Orbiter cant leave this round
+		OrbiterCantLeaveThisRound,
+		/// This orbiter has not made a deposit
+		OrbiterDepositNotFound,
+		/// This orbiter is not found
+		OrbiterNotFound,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Paid the orbiter account the balance as liquid rewards.
+		OrbiterRewarded {
+			account: T::AccountId,
+			rewards: BalanceOf<T>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Add an orbiter in a collator pool
+		#[pallet::weight(500_000_000)]
+		pub fn collator_add_orbiter(
+			origin: OriginFor<T>,
+			orbiter: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let collator = ensure_signed(origin)?;
+			let orbiter = T::Lookup::lookup(orbiter)?;
+
+			let mut collator_pool =
+				CollatorsPool::<T>::get(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+			let orbiters = collator_pool.get_orbiters();
+			ensure!(
+				(orbiters.len() as u32) < T::MaxPoolSize::get(),
+				Error::<T>::CollatorPoolTooLarge
+			);
+			for orbiter_ in orbiters {
```

Comment:
maybe using orbiters.iter().find() reads better?

Reply:
Yes, any() is even better

Reply:
Done